src/status_im2/contexts/quo_preview/share/share_qr_code.cljs

```diff
@@ -1,34 +1,141 @@
 (ns status-im2.contexts.quo-preview.share.share-qr-code
-  (:require
-    [quo.core :as quo]
-    [reagent.core :as reagent]
-    [status-im2.contexts.quo-preview.preview :as preview]
-    [utils.image-server :as image-server]
-    [utils.re-frame :as rf]))
+  (:require [quo.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.common.resources :as resources]
+            [status-im2.contexts.quo-preview.preview :as preview]
+            [utils.image-server :as image-server]
+            [utils.re-frame :as rf]))
 
 (def descriptor
-  [{:key :url :type :text}
-   {:key :link-title :type :text}])
+  [{:key   :qr-data
+    :label "QR data:"
+    :type  :text}
+   {:key     :type
+    :type    :select
+    :options [{:key :profile}
+              {:key :wallet-legacy}
+              {:key :wallet-multichain}]}])
```

Comment:
I think it would be better if there is 2 types `profile` and `wallet`, and then when wallet is selected, `legacy` and `multichain` should be handled from within the component. When using the component in flows it will be either the `profile` variant or the `wallet` variant (which defaults to `legacy` and the user can change it to `multichain`)

Reply:
Hey @OmarBasem 

We agreed on reflect the figma properties as mucha as we can.

In figma, the component receives a property `type` and the possible types are the ones I listed:
![image](https://github.com/status-im/status-mobile/assets/90291778/497ee86e-c839-4ee9-8b25-5a9ecb5ed728)

![image](https://github.com/status-im/status-mobile/assets/90291778/29738432-a19e-4a31-9b49-ce8b9a1c0b31)
![image](https://github.com/status-im/status-mobile/assets/90291778/72cff82d-e98e-480e-8487-30dcaf299aeb)

![image](https://github.com/status-im/status-mobile/assets/90291778/3c70ba0a-e60b-49b8-bfe8-db04c8d36904)


> Currently in the preview screen I am not able to switch between legacy and multichain by tapping the buttons on the component.

Those tabs receive a callback, that callback could be used to mutate the a ratom that triggers a `:type` change :+1: